src/ai/backend/manager/data/permission/role.py

```diff
@@ -74,11 +76,41 @@ class RoleDataWithPermissions:
 
 
 @dataclass
-class PermissionCheckInput:
+class MergedPermissionData:
+    scope_permissions: Mapping[ScopeId, set[OperationType]]
+    global_permissions: set[OperationType]
+    object_permissions: Mapping[ObjectId, set[OperationType]]
```

Comment:
It seems like it would be good to separate it. However, I think we need to consider how to separate it a little bit, so I'll take a look and suggest some ideas.